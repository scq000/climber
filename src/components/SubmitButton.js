import React from 'react';
import {View, TouchableOpacity, StyleSheet, Text, Image} from 'react-native';

const SubmitButton = ({onSubmit, disabled}) => {
  return (
    <TouchableOpacity onPress={onSubmit}>
      <View style={[styles.button, {opacity: disabled ? 0.5 : 1}]}>
        <Image source={require('../assets/btn/bg.png')} style={styles.image} />
        <View style={styles.absoluteView}>
          <Text style={styles.text}>LOGIN</Text>
        </View>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    width: 312,
    height: 39.19,
    borderColor: 'red',
    justifyContent: 'center',
    alignItems: 'center',
    marginVertical: 23.5,
  },
  absoluteView: {
    flex: 1,
    position: 'absolute',
    alignItems: 'center',
    justifyContent: 'center',
  },
  text: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    color: '#fff',
    fontWeight: 'bold',
  },
  image: {
    borderRadius: 19,
  },
});

export default SubmitButton;
