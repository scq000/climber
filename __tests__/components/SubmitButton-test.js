import React from 'react';
import 'react-native';
import {create} from 'react-test-renderer';
import SubmitButton from '../../src/components/SubmitButton';

describe('SubmitButton', () => {
  const fn = jest.fn();

  it('should render correctly', function () {
    const snap = create(<SubmitButton onSubmit={fn} />).toJSON();
    expect(snap).toMatchSnapshot();
  });

  it('should render disabled button correctly', function () {
    const snap = create(<SubmitButton onSubmit={fn} disabled />).toJSON();
    expect(snap).toMatchSnapshot();
  });
});
