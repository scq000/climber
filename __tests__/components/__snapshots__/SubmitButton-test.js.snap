// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitButton should render correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderColor": "red",
          "height": 39.19,
          "justifyContent": "center",
          "marginVertical": 23.5,
          "width": 312,
        },
        Object {
          "opacity": 1,
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "process": [Function],
        }
      }
      style={
        Object {
          "borderRadius": 19,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "position": "absolute",
        }
      }
    >
      <Text
        style={
          Object {
            "alignItems": "center",
            "color": "#fff",
            "flex": 1,
            "fontWeight": "bold",
            "justifyContent": "center",
          }
        }
      >
        LOGIN
      </Text>
    </View>
  </View>
</View>
`;

exports[`SubmitButton should render disabled button correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderColor": "red",
          "height": 39.19,
          "justifyContent": "center",
          "marginVertical": 23.5,
          "width": 312,
        },
        Object {
          "opacity": 0.5,
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "process": [Function],
        }
      }
      style={
        Object {
          "borderRadius": 19,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "position": "absolute",
        }
      }
    >
      <Text
        style={
          Object {
            "alignItems": "center",
            "color": "#fff",
            "flex": 1,
            "fontWeight": "bold",
            "justifyContent": "center",
          }
        }
      >
        LOGIN
      </Text>
    </View>
  </View>
</View>
`;
